set.mm

```diff
@@ -485570,6 +485603,100 @@ commutative monoid (=vectors) together with a semiring (=scalars) and a
   $}
 
 
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  Field extensions generated by a set
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  $c fldGen $.
+
+  $( Syntax for a function generating sub-fields. $)
+  cfldgen $a class fldGen $.
+
+  ${
+    $d f s a $.
+    $( Define a function generating the smallest sub-division-ring containing a
+       given set.  If the base structure is a division ring, this is also a
+       division ring (see ~ fldgensdrg ).  If the base structure is a field,
+       this is a subfield (see ~ fldgenfld and ~ fldsdrgfld ).  In general this
+       will be used in the context of fields, therefore the name ` fldGen ` .
+       (Contributed by Saveliy Skresanov and Thierry Arnoux, 9-Jan-2025.) $)
+    df-fldgen $a |- fldGen = ( f e. _V , s e. _V |->
+                                   |^| { a e. ( SubDRing ` f ) | s C_ a } ) $.
+  $}
+
+  ${
+    $d B a $.  $d F a f s x $.  $d S a f s x $.  $d ph a x $.
+    fldgenval.1 $e |- B = ( Base ` F ) $.
+    fldgenval.2 $e |- ( ph -> F e. DivRing ) $.
+    ${
+      fldgenval.3 $e |- ( ph -> S C_ B ) $.
+      $( Value of the field generating function.  (Contributed by Thierry
+         Arnoux, 11-Jan-2025.) $)
+      fldgenval $p |- ( ph
```

Comment:
Is ` fldgenval.3 $e |- ( ph -> S C_ B ) $. ` required for ~fldgenval? I think ` fldgenval.3 $e |- ( ph -> S e. V ) $.` (sethood) would be sufficient, wouldn't it?

Maybe the meaning of the classes should be explained in the comment, e.g.. 

```
` ( F fldGen S ) ` is the smallest sub-division-ring of ` F ` containing ` S `.
```

Reply:
> Is `fldgenval.3 $e |- ( ph -> S C_ B ) $.` required for ~fldgenval? 

It is actually required. The function value ``|^| { a e. ( SubDRing ` F ) | S C_ a }`` has to be proven to exist, and for that, by [`intex`](https://us.metamath.org/mpeuni/intex.html), we need ``{ a e. ( SubDRing ` F ) | S C_ a }`` to be non empty. This is the case because ``B e. ( SubDRing ` F )`` and `S C_ B`.

Reply:
> Maybe the meaning of the classes should be explained in the comment

Comment updated!